docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,81 @@
+---
+title: add-discovery-kernel-arguments
+authors:
+- "@oamizur"
+  creation-date: 2022-09-20
+  last-updated: 2022-09-20
+---
+
+# Add discovery kernel arguments
+
+## Summary
+
+Some environments require additional kernel arguments to boot host machines.  
+We would like to support additional kernel arguments while booting the discovery image.
+
+## Motivation
+
+Previously, adding kernel arguments was required by several customers in order to adapt to various environments.
+For example **rd.net.timeout.carrier** may be required in order for either static networking to be applied or 
+to successfully get a dhcp address before pulling the rootfs.
+### Goals
+Add capability to set additional kernel arguments to discovery image.
+
+Support the following scenarios:
+- Full ISO image
+- Minimal ISO image
+- iPXE
+- Cloud/operator (when considering authentication from image service to assisted-service)
+
+### Non-Goals
+Manipulation of existing kernel parameters (update/delete)
+
+## Proposal
+
+The discovery kernel arguments will be stored in the infra-env database record in a designated JSON encoded field.
+During infra-env registration and update the full set of the kernel arguments can be set.
+
+Setting the arguments can either be done by either REST API or Kube API (Part of InfraEnv CR)
+### User Stories
+
+#### Story 1
+
+When having slow NIC, increase the timeout to have the IP address configured.
+
+### Implementation Details/Notes/Constraints
+
+#### iPXE implementation
+
+iPXE implementation should be quite straight forward.  The arguments will be concatenated at the end of the 
+**kernel** command which is part of the iPXE script.
+
+#### Minimal and full ISO
+
+The embedding of the kernel arguments into the ISO is an issue that needs to be investigated.
+
+We will also might need the image service to query the assisted service to get the kernel arguments.
+This will require authentication.  The authentication methodology may have 2 variants:  
```

Comment:
The image service already queries the assisted service to get the ignition info

Reply:
OK